source/common/ssl/context_impl.cc

```diff
@@ -12,18 +12,31 @@
 #include "common/common/hex.h"
 
 #include "fmt/format.h"
+#include "openssl/rand.h"
 #include "openssl/x509v3.h"
 
 namespace Envoy {
 namespace Ssl {
 
 const unsigned char ContextImpl::SERVER_SESSION_ID_CONTEXT = 1;
+int ContextImpl::ssl_context_index_ = -1;
 
 ContextImpl::ContextImpl(ContextManagerImpl& parent, Stats::Scope& scope, ContextConfig& config)
     : parent_(parent), ctx_(SSL_CTX_new(TLS_method())), scope_(scope),
       stats_(generateStats(scope)) {
   RELEASE_ASSERT(ctx_);
 
+  if (ssl_context_index_ < 0) {
+    ssl_context_index_ = SSL_CTX_get_ex_new_index(0, nullptr, nullptr, nullptr, nullptr);
+    if (ssl_context_index_ < 0) {
```

Comment:
drive by review comment: Please see https://github.com/envoyproxy/envoy/blob/master/STYLE.md#error-handling for guidance on error handling. IMO a lot of the error handling added in this PR should be crashing asserts. (Obviously anything that involves loading a ticket file, etc. should have normal error handling). Will defer to @PiotrSikora on which should be which.

Reply:
That's a really good point.  What's the preferred way to handle that?  RELEASE_ASSERT()? PANIC()?

Reply:
Typically we use RELEASE_ASSERT() so it can all be inline.